enigma-swing/src/main/java/cuchaz/enigma/gui/panels/StructurePanel.java

```diff
@@ -18,17 +20,38 @@
 import java.awt.event.MouseEvent;
 
 public class StructurePanel extends JPanel {
-    private JPanel sortingPanel;
-    private JCheckBox hideDeobfuscated;
+    private final JPanel optionsPanel;
 
-    private JTree structureTree;
+    private final JLabel obfuscationVisibilityLabel = new JLabel();
+    private final JLabel documentationVisibilityLabel = new JLabel();
+    private final JLabel sortingOrderLabel = new JLabel();
+
+    private final JComboBox<StructureTreeOptions.ObfuscationVisibility> obfuscationVisibility;
+    private final JComboBox<StructureTreeOptions.DocumentationVisibility> documentationVisibility;
+    private final JComboBox<StructureTreeOptions.SortingOrder> sortingOrder;
+
+    private final JTree structureTree;
 
     public StructurePanel(Gui gui) {
-        this.sortingPanel = new JPanel();
-        this.hideDeobfuscated = new JCheckBox(I18n.translate("info_panel.tree.structure.hide_deobfuscated"));
-        this.hideDeobfuscated.addActionListener(event -> gui.showStructure(gui.getActiveEditor()));
-        this.sortingPanel.add(this.hideDeobfuscated);
-        this.sortingPanel.setVisible(false);
+        this.optionsPanel = new JPanel(new GridBagLayout());
+        this.optionsPanel.setVisible(false);
+
+        GridBagConstraintsBuilder cb = GridBagConstraintsBuilder.create().insets(5).fill(GridBagConstraints.HORIZONTAL);
+
+        this.optionsPanel.add(this.obfuscationVisibilityLabel, cb.pos(0, 0).build());
+        this.obfuscationVisibility = new JComboBox<>(StructureTreeOptions.ObfuscationVisibility.values());
```

Comment:
If you don't use toString, seems you can utilize the combo box editor of the j combo box.

Reply:
I'm not exactly sure what you mean, could you explain?